src/linear_algebra/tensor_product.lean

```diff
@@ -805,6 +805,33 @@ rfl
   (congr f g).symm (p ⊗ₜ q) = f.symm p ⊗ₜ g.symm q :=
 rfl
 
+variables (R M N P Q)
+
+/-- This special case is worth defining explicitly since it is useful for defining multiplication
+on tensor products of modules carrying multiplications (e.g., associative rings, Lie rings, ...).
+
+E.g., suppose `M = P` and `N = Q` and that `M` and `N` carry bilinear multiplications:
+`M ⊗ M → M` and `N ⊗ N → N`. Using `map`, we can define `(M ⊗ M) ⊗ (N ⊗ N) → M ⊗ N` which, when
+combined with this definition, yields a bilinear multiplication on `M ⊗ N`:
+`(M ⊗ N) ⊗ (M ⊗ N) → M ⊗ N`. -/
+def comm_two_in_four : (M ⊗[R] N) ⊗[R] (P ⊗[R] Q) ≃ₗ[R] (M ⊗[R] P) ⊗[R] (N ⊗[R] Q) :=
+let e₁ := tensor_product.assoc R M N (P ⊗[R] Q),
+    e₂ := congr (1 : M ≃ₗ[R] M) (tensor_product.assoc R N P Q).symm,
+    e₃ := congr (1 : M ≃ₗ[R] M) (congr (tensor_product.comm R N P) (1 : Q ≃ₗ[R] Q)),
+    e₄ := congr (1 : M ≃ₗ[R] M) (tensor_product.assoc R P N Q),
+    e₅ := (tensor_product.assoc R M P (N ⊗[R] Q)).symm in
+(((e₁.trans e₂).trans e₃).trans e₄).trans e₅
```

Comment:
The meaning is slightly different, but `trans` is associative and it maybe reads more cleanly:
```suggestion
e₁.trans $ e₂.trans $ e₃.trans $ e₄.trans e₅
```